app/controllers/welcome_controller.rb

```diff
@@ -1,4 +1,7 @@
 class WelcomeController < ApplicationController
+  before_action :authenticate_user!
   def index
+    @posts = Post.all.reverse
```

Comment:
this looks like the same as the posts controller? Why not just set root to be /posts ?